lib/web/sign.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"crypto/x509/pkix"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+	"golang.org/x/exp/slices"
```

Comment:
Looks like this is imported for the "contains" function, can we use `utils.StringInSlice` that we already have instead?

Reply:
`utils.SliceContainsStr` from `github.com/gravitational/teleport/api/utils` right?

The issue here is that our variables are of type `identityfile.Format` and not a `string`
So, I would probably need to convert things around between both formats

Reply:
Yeah, I gotcha. I think it's ok to keep it then. We have to start using templates sooner or later :)